Code/Framework/AzCore/AzCore/EBus/Internal/CallstackEntry.h

```diff
@@ -161,33 +161,43 @@ namespace AZ
         template <class C>
         struct EBusCallstackStorage<C, true>
         {
-            static AZ_THREAD_LOCAL C* s_entry;
-
             EBusCallstackStorage() = default;
             ~EBusCallstackStorage() = default;
             EBusCallstackStorage(const EBusCallstackStorage&) = delete;
             EBusCallstackStorage(EBusCallstackStorage&&) = delete;
 
             C* operator->() const
             {
-                return s_entry;
+                return GetEntry();
             }
 
             C& operator*() const
             {
-                return *s_entry;
+                return *GetEntry();
             }
 
             C* operator=(C* entry)
             {
-                s_entry = entry;
-                return s_entry;
+                GetEntry() = entry;
+                return GetEntry();
             }
 
             operator C*() const
             {
-                return s_entry;
+                return GetEntry();
             }
+
+        private:
+            C*& GetEntry() const;
         };
+
+        // This functino needs to be defined outside of the class definition such that it can get explicitly instantiated correctly. This is
```

Comment:
Spelling `functino` (sorry for the nit!)

Reply:
I wish I knew more about the internals of ebus - I've always wanted to ask the question, what is the callstack storage for anyway?  is it necessary to function or just a debug tool?  if the latter, could we remove it except in debug or something?  NotE:  NOT NECESSARY to change this for this PR, just curious as to whether you have any insight for future.  It costs a bit of overhead and constantly shows up when I use a binary analysis tool to find whats inside a binary.